QualificationChecker/Controllers/CandidatesController.cs

```diff
@@ -11,100 +13,63 @@ public partial class CandidatesController : ControllerBase
 
-                // only run this if they pass the knockout questions
-                if (candidate.OrganizationId == request.Organization.OrgId && isInterested && meetsAgeRequirements)
+                else
                 {
-                    // Create a dictionary to store organization answers by PositionId and Id
-                    var orgAnswersDictionary = request.Organization.OrgQuestions
-                        .GroupBy(q => q.PositionId)
-                        .ToDictionary(group => group.Key, group => group.ToList());
+                    //Loop through each candaite and compare answers
+                    foreach (var candidate in request.Candidates)
 
-                    // Initialize a flag to track if all answers match
-                    bool allAnswersMatch = true;
+                    {   // Initialize you Candidate to work with utilities
+                        var myCandidate = new CandidateUtilities(candidate, request.Organization);
 
-                    // Loop through each candidate question
-                    foreach (var candidateAnswer in candidate.CandidateQuestions)
-                    {
-                        // Create a lookup key to match the PositionId
-                        var lookupKey = candidateAnswer.PositionId;
 
-                        if (orgAnswersDictionary.TryGetValue(lookupKey, out var OrgAnswers))
+                        // only run this if they pass the knockout questions
+                        if (myCandidate.Candidate.OrganizationId == myCandidate.Organization.OrgId && myCandidate.IsInterested() && myCandidate.MeetsAgeRequirement())
                         {
-                            // Loop through the orgAnswers (in case there are multiple with the same PositionId)
-                            bool answersMatch = false;
 
-
-                            foreach (var OrgAnswer in OrgAnswers)
+                            // Check to make sure postionId makes a dictionary 
+                            if (myCandidate.OrgAnswersDictionary != null)
                             {
-                                // make sure they have the same number first before checking answers
-                                if (OrgAnswers.Count != candidate.CandidateQuestions.Count)
-                                {
-                                    answersMatch = false;
-                                    break;
-                                }
-
-                                if (!MatchAnswers(candidateAnswer.Answer, OrgAnswer.Answer))
-                                {
-                                    // If an answer doesn't match or orgAnswer is null, set answersMatch to false
-                                    answersMatch = false;
-                                    break;
-                                }
-                                answersMatch = true;
+                                QualifiedCandidates.AddRange(myCandidate.CheckQuestions());
```

Comment:
Surprised a method called `CheckQuestions` would return a candidate.  Would expect some sort of validation from the check method.